orangewidget/utils/listview.py

```diff
@@ -10,9 +11,19 @@
     QSortFilterProxyModel,
     QItemSelection,
     QSize,
+    QItemSelectionModel,
 )
 
 
+@contextmanager
+def disconnected(signal, slot, connection_type=Qt.AutoConnection):
+    signal.disconnect(slot)
+    try:
+        yield
+    finally:
+        signal.connect(slot, connection_type)
```

Comment:
Perhaps blocking signals (https://doc.qt.io/qt-6/qobject.html#blockSignals) would be better the temporarily disconnecting them?

Also: search the code for `def blocked(`. You'll find that a similar function already exists in two other places. Perhaps we should introduce one in `orange-widget-base` and then (after release...) use it wherever needed?

Reply:
Or even use an existing one at 
https://github.com/biolab/orange-widget-base/blob/6ddb801d2467d2f2970316b99100fb15df2d911e/orangewidget/utils/itemmodels.py#L43